src/watchdog/watchmedo.py

```diff
@@ -632,14 +640,33 @@ def handler_termination_signal(_signum, _frame):
         handler.stop()
 
 
+def _get_log_level_from_args(args):
+    verbosity = sum(args.verbosity or [])
+    if verbosity < -1:
+        raise Exception("-q/--quiet may be specified only once.")
+    if verbosity > 2:
+        raise Exception("-v/--verbose may be specified up to 2 times.")
+    return ['ERROR', 'WARNING', 'INFO', 'DEBUG'][1 + verbosity]
+
+
 def main():
     """Entry-point function."""
     args = cli.parse_args()
     if args.top_command is None:
         cli.print_help()
-    else:
-        args.func(args)
+        return 1
+
+    try:
+        log_level = _get_log_level_from_args(args)
+    except Exception as exc:
+        print("Error: " + exc.args[0], file=sys.stderr)
+        command_parsers[args.top_command].print_help()
```

Comment:
I think one thing is missing: where did you populate `command_parsers`?

```bash
$ watchmedo log -vvv\
    --patterns="*.py;*.txt" \
    --ignore-directories \
    --recursive \
    .
```
```python
Error: -v/--verbose may be specified up to 2 times.
Traceback (most recent call last):
  File "src/watchdog/watchmedo.py", line 660, in main
    log_level = _get_log_level_from_args(args)
  File "src/watchdog/watchmedo.py", line 648, in _get_log_level_from_args
    raise Exception("-v/--verbose may be specified up to 2 times.")
Exception: -v/--verbose may be specified up to 2 times.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "venv3.9.8/bin/watchmedo", line 33, in <module>
    sys.exit(load_entry_point('watchdog', 'console_scripts', 'watchmedo')())
  File "src/watchdog/watchmedo.py", line 663, in main
    command_parsers[args.top_command].print_help()
KeyError: 'log'
```

Reply:
And maybe should we raise a custom exception rather than the broad `Exception`. WDYT?

Reply:
Whoops, good catch! That was indeed missing, not sure how it got dropped.